src/exec.c

```diff
@@ -408,21 +414,65 @@ static int fork_child(program_list_t *pl, int *fd_in, int *fd_out,
         grbuf_size = sysconf(_SC_PAGESIZE);
       if (grbuf_size <= 0)
         grbuf_size = 4096;
-      char grbuf[grbuf_size];
 
-      status = getgrnam_r(pl->group, &gr, grbuf, sizeof(grbuf), &gr_ptr);
-      if (status != 0) {
-        ERROR("exec plugin: Failed to get group information "
-              "for group ``%s'': %s",
-              pl->group, STRERROR(status));
-        goto failed;
+      long int size;
```

Comment:
The code doing the group name to ID lookup is getting quite long and blowing up the `fork_child()` function. Could you please move that code (essentially everything in this block) to its own function?

Reply:
Hi @octo , 
I have made the change to refactor this block of group ID lookup code into a separate function. Please review.